core/src/main/java/org/elasticsearch/index/query/TermsQueryBuilder.java

```diff
@@ -30,96 +54,209 @@
-     * @param name   The field name
+     * @param fieldName The field name
      * @param values The terms
      */
-    public TermsQueryBuilder(String name, int... values) {
-        this.name = name;
-        this.values = values;
+    public TermsQueryBuilder(String fieldName, int... values) {
+        this(fieldName, values != null ? Ints.asList(values) : null);
     }
 
     /**
      * A filter for a field based on several terms matching on any of them.
      *
-     * @param name   The field name
+     * @param fieldName The field name
      * @param values The terms
      */
-    public TermsQueryBuilder(String name, long... values) {
-        this.name = name;
-        this.values = values;
+    public TermsQueryBuilder(String fieldName, long... values) {
+        this(fieldName, Longs.asList(values));
     }
 
     /**
      * A filter for a field based on several terms matching on any of them.
      *
-     * @param name   The field name
+     * @param fieldName The field name
      * @param values The terms
      */
-    public TermsQueryBuilder(String name, float... values) {
-        this.name = name;
-        this.values = values;
+    public TermsQueryBuilder(String fieldName, float... values) {
+        this(fieldName, Floats.asList(values));
     }
 
     /**
      * A filter for a field based on several terms matching on any of them.
      *
-     * @param name   The field name
+     * @param fieldName The field name
      * @param values The terms
      */
-    public TermsQueryBuilder(String name, double... values) {
-        this.name = name;
-        this.values = values;
+    public TermsQueryBuilder(String fieldName, double... values) {
+        this(fieldName, Doubles.asList(values));
     }
 
     /**
      * A filter for a field based on several terms matching on any of them.
      *
-     * @param name   The field name
+     * @param fieldName The field name
      * @param values The terms
      */
-    public TermsQueryBuilder(String name, Object... values) {
-        this.name = name;
-        this.values = values;
+    public TermsQueryBuilder(String fieldName, Object... values) {
+        this(fieldName, values != null ? Arrays.asList(values) : null);
+    }
+
+    /**
+     * Constructor used for terms query lookup.
+     *
+     * @param fieldName The field name
+     */
+    public TermsQueryBuilder(String fieldName) {
+        this.fieldName = fieldName;
     }
 
     /**
      * A filter for a field based on several terms matching on any of them.
      *
-     * @param name   The field name
+     * @param fieldName The field name
      * @param values The terms
      */
-    public TermsQueryBuilder(String name, Iterable values) {
-        this.name = name;
-        this.values = values;
+    public TermsQueryBuilder(String fieldName, Iterable values) {
+        this.fieldName = fieldName;
+        if (values != null) {
+            this.values.addAll(convertToBytesRefListIfStringList(values));
+        }
```

Comment:
Two concerns here: If we convert Strings to BytesRef here, shouldn't we also do so in the (..., String... values) constructor? Also, since this seems a convenience method, I'd prefer converting the content of the Iterable to be converted to some array and then use the (..., Object...) constructor, as there are already too many ways of constructing this query builder.